### ISSUE TITLE
[4.x]: Cart is wiped out when logging in

### ISSUE THREAD

**Issue by elivz** [craftcms/commerce#3445]
### Description
It appears that the intended logic that only restored a user's previous cart when logging in if their current cart is empty has gone missing at some point. It is still referenced in a [code comment](https://github.com/craftcms/commerce/blob/f58b25746f573e3064a92daa8b272897b2e66f1b/src/services/Carts.php#L328), but I don't see anywhere that would actually implement that. As a result, when logging in after adding items to my cart, the cart is emptied.

### Steps to reproduce

1. Without logging in, add some items to your shopping cart
2. Log into an existing user account that does not have a previous saved cart

### Expected behavior

* Line items added before logging in should still be present in the cart.

### Actual behavior

* The cart is empty

### Craft CMS version

4.8.6

### Craft Commerce version

4.5.3

### PHP version

8.1

**Comment by CrystalHolmes** [comment]
any updates on this @lukeholder ?

**Comment by nfourtythree** [comment]
Hi All

Thank you for bringing this to our attention. We have pushed a fix which will be included in the next release of Commerce.

To get this early, change your `craftcms/commerce` requirement in your project's `composer.json` to:

```json
"require": {
  "craftcms/commerce": "dev-develop#e3f9983484db2bb9f7019d6a06fe24ecd4474909 as 4.5.3",
  "...": "..."
}
```

Then run `composer update`.

Thanks!

**Comment by nfourtythree** [comment]
This fix has now be released as part of Commerce [4.5.4](https://github.com/craftcms/commerce/releases/tag/4.5.4)

Thanks!
